src/libertem/win_tweaks.py

```diff
@@ -47,10 +58,13 @@ def set_console_mode(mode, stream=sys.stdin):
 
 
 def disable_quickedit():
-    mode = get_console_mode()
-    mode &= ~ENABLE_QUICK_EDIT
-    mode |= ENABLE_EXTENDED_FLAGS
-    set_console_mode(mode)
+    try:
+        mode = get_console_mode()
+        mode &= ~ENABLE_QUICK_EDIT
+        mode |= ENABLE_EXTENDED_FLAGS
+        set_console_mode(mode)
+    except TypeError:
+        logger.error("Incompatible types!")
 
```

Comment:
Instead of reacting to `TypeError`, what about catching the original `Exception` here, that is, the `ctypes.WinError`? In the `get_console_mode` function, if you are catching the exception, it is not clear what to do in case of an error, because people could be calling the function for whatever reason. That's why it is throwing an exception in the first place! In `disable_quickedit`, you know what to do: ignore the error, because the console likely doesn't have quickedit functionality, and log a message accordingly.

A note on the log messages: "Incompatible types" and "Failed!!!" are not very helpful: imagine a user reading these, not knowing where they come from. They would be alarmed! But in reality, most likely, it is just a side-effect of them using git bash. The message should be clear and describe what is happening.

The log severity should reflect this, too: not disabling quick edit on git bash should not be a cause for concern, it should be logged as `DEBUG` (or `INFO` at most) message, in case this ever fails outside of git bash for some reason.

Reply:
Okay, I will try to work on the `ctypes.WinError` and make the other changes as well.